src/libs/actions/BankAccounts.js

```diff
@@ -314,115 +315,131 @@ function goToWithdrawalAccountSetupStep(stepID, achData) {
-            bankAccountListResponse,
-        ]) => {
-            const bankAccountID = lodashGet(freePlanBankAccountIDResponse, [
-                'value', 'nameValuePairs', CONST.NVP.FREE_PLAN_BANK_ACCOUNT_ID,
-            ], '');
-            const kycVerificationsMigration = lodashGet(kycVerificationsMigrationResponse, [
-                'value', 'nameValuePairs', 'expensify_migration_2020_04_28_RunKycVerifications',
-            ], '');
-            const throttledDate = lodashGet(achDataThrottledResponse, [
-                'value', 'nameValuePairs', CONST.NVP.ACH_DATA_THROTTLED,
+    let bankAccountID;
+    let failedValidationAttemptsName;
+
+    API.Get({
+        returnValueList: 'nameValuePairs',
+        name: CONST.NVP.FREE_PLAN_BANK_ACCOUNT_ID,
+    })
+        .then((response) => {
+            bankAccountID = lodashGet(response, ['nameValuePairs', CONST.NVP.FREE_PLAN_BANK_ACCOUNT_ID,
             ], '');
-            const bankAccountJSON = _.find(
-                lodashGet(bankAccountListResponse, ['value', 'bankAccountList'], []), account => (
-                    account.bankAccountID === bankAccountID
-                ),
-            );
-            const bankAccount = bankAccountJSON ? new BankAccount(bankAccountJSON) : null;
-
-            // Next we'll build the achData and save it to Onyx
-            // If the user is already setting up a bank account we will continue the flow for them
-            let currentStep = previousACHData.currentStep;
-            const achData = bankAccount ? bankAccount.toACHData() : {};
-            achData.useOnfido = true;
-            achData.policyID = '';
-            achData.isInSetup = !bankAccount || bankAccount.isInSetup();
-            achData.bankAccountInReview = bankAccount && bankAccount.isVerifying();
-            achData.domainLimit = 0;
-            achData.isDomainUsingExpensifyCard = false; // @TODO - Not actually sure if we need this
-
-            // @TODO This subStep is used to either show the Plaid "login" view or the "manual" view - but not sure if
-            // we need to implement it yet...
-            // eslint-disable-next-line max-len
-            // See Web-Secure: https://github.com/Expensify/Web-Expensify/blob/896941794f68d7dce64466d83a3e86a5f8122e45/site/app/settings/reimbursement/bankAccountView.jsx#L356-L357
-            achData.subStep = '';
-
-            // If we're not in setup, it means we already have a withdrawal account and we're upgrading it to a business
-            // bank account. So let the user review all steps with all info prefilled and editable, unless a specific
-            // step was passed.
-            if (!achData.isInSetup) {
-                // @TODO Not sure if we need to do this since for NewDot none of the accounts are pre-existing ones
-                currentStep = '';
-            }
+            failedValidationAttemptsName = CONST.NVP.FAILED_BANK_ACCOUNT_VALIDATIONS_PREFACE + bankAccountID;
```

Comment:
NAB, I think this can be 

```js
const failedValidationAttemptsName = ...
```` 

since the other usage is in the same scope. Unless we do something like this...

```js
    return promiseAllSettled(...);
})
    .then(() => {
        ...
    });   
```

Which is a pattern we use sometimes when chaining promises.